### PR TITLE
Contribute: Various fixes from TV 2

### PR DESCRIPTION
**Pull request by thomasslee97**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
A variety of fixed are included in this PR:
- Copy adLibSourceId to continuation pieces for an infinite
  - Fixes a bug where continuation pieces of an infinite would not have the adLibSourceId copied to them, which is required for "toggling" and infinite in dashboards.
- chore: use cache for expected playout items for adLibs
  - Previously this would make a direct database call, rather than the cache. This has no performance benefit right now, but is more consistent and may mean that a performance improvement is seen in the future with continued work on the caching system.
- fix: Standalone shelf route
  - The routing for standalone shelf views had been lost at some point, this restores the missing route.
- fix: arrow overlapping with extra time label
  - Small visual fix to the positioning of the overflow time label.
- fix: Crop WithinPart piece instances
  - There was a difference between what the UI was expecting (a unix timestamp) and what the data in the database contained (time relative to the start of the part). This resulted in pieces with WithinPart lifespans not getting cropped correctly when stopped (e.g. by stopping pieces on a sourcelayer).
- Set definitely ended to 1 second
  - With some experimentation at TV 2, we have found that the current definitely ended time (10 seconds) lets pieces live for too long, impacting performance. We have been testing with this value since 02/09 and have not found any issues created by lowering this value.
- fix: Don't push group name to used hotkeys
  - The hotkey group name was being pushed to usedHotkeys in dashboard panels, for every hotkey.
- fix: Dashboard crash
  - It's possible for props.layer to be undefined, so this PR changes it to be optional to enforce checking for the presence of layer.
- fix: Subscribe to all parts, pieces, and instances if in standalone shelf
  - fix: Only fetch piece instances for current, next, and previous part
  - Two commits. Previously, the standalone shelf wouldn't fetch any data, meaning it can't perform actions like stopping pieces and much of the context-aware functionality of the shelf was broken. These two commits mean that the standalone shelf will fetch data from the current, next, and previous parts; which has been enough data so far from testing.
- fix: timed pieces showing outside of their part
  - In cases where you have a part that is set to say 10 seconds, which contain pieces that start after 15 seconds (graphics for example), these graphics would appear outside of the part (often over other parts). This change applies CSS to hide these pieces until the part is played out long enough for them to be about to play out.
- fix: store playlist id in expected playout items
  - This adds playlistId property to ExpectedPlayoutItem because Playout Gateway and TSR need it to filter items that belong to the active rundown playlist

**Status**

- [X] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

**Comment by nytamin** [comment]
status: I'm going to ask @jstarpl and @Julusian to review this as well, when all 3 of us are happy we'll merge this in.

**Review by Julusian — APPROVED** [review]
For the bits I know anything about, I am happy.
Not 100% sure about DEFINITELY_ENDED_FUTURE_DURATION being 1 second, rather than a few seconds, but I am happy for that to be a testing note and we can increase it if it is a problem. The 10s was an arbitrary 'safe' number after all

**Review by nytamin — APPROVED** [review]

**Review by jstarpl — APPROVED** [review]
Erroneously marked this as approved. I think it would be best to keep the solution in place in R25 in SourceLayerItem.tsx:325 and I think the lines around RundownView.tsx:1575 need to be checked if they actually work.

**Review by jstarpl — CHANGES_REQUESTED** [review]
I realized that my previous comments have not been addressed.

File: meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx [hunk 1]
```diff
@@ -321,8 +321,11 @@ export const SourceLayerItem = withTranslation()(
 			const expectedDurationNumber =
 				typeof innerPiece.enable.duration === 'number' ? innerPiece.enable.duration || 0 : 0
 			const userDurationNumber =
-				piece.instance.userDuration && typeof piece.instance.userDuration.end === 'number' && innerPiece.startedPlayback
-					? piece.instance.userDuration.end - innerPiece.startedPlayback
+				piece.instance.userDuration &&
+				typeof piece.instance.userDuration.end === 'number' &&
+				innerPiece.startedPlayback &&
+				innerPiece.stoppedPlayback
+					? innerPiece.stoppedPlayback - innerPiece.startedPlayback
```
[the remark this answers is not in the document]
Comment (thomasslee97) [reply meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx:328]: It looks like this block of code was removed from r24 at least, so this fix is probably not relevant https://github.com/nrkno/tv-automation-server-core/commit/3001a044271ae10141bb68d854ea5ef0296276bf

File: meteor/client/ui/RundownView.tsx [hunk 2]
```diff
@@ -1567,6 +1568,23 @@ export const RundownView = translateWithTracker<IProps, IState, ITrackedProps>((
...
 						},
 					})
+					if (this.props.onlyShelf) {
+						this.subscribe(PubSub.pieceInstances, {
+							rundownId: {
+								$in: rundownIDs,
+							},
+							partInstanceId: {
+								$in: [
+									playlist.currentPartInstanceId,
```
[the remark this answers is not in the document]
Comment (thomasslee97) [reply meteor/client/ui/RundownView.tsx:1596]: I have moved it to its own autorun here https://github.com/olzzon/tv-automation-server-core/commit/4ccdcb58df42bfdf4e9dbafd08f0ac15168a8be1

[the remark this answers is not in the document]
Comment (thomasslee97) [reply meteor/client/ui/RundownView.tsx:1586]: And reduced data fetched, seeing as we don't need to fetch the rundown Ids given that the part instance Ids should be unique https://github.com/olzzon/tv-automation-server-core/commit/d474f5ebe9ccf4d9aa675c084b0f508eee97b70e

File: meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomasslee97) [reply meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx]: I've dropped the commit that produced this code in this PR

[the remark this answers is not in the document]
Comment (jstarpl) [reply meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx]: OK, I'm closing this then 👍

File: meteor/client/ui/RundownView.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jstarpl) [reply meteor/client/ui/RundownView.tsx:1586]: Great 👍

**Review by jstarpl — APPROVED** [review]
Looks good!

**Pull request merged**